partner_compassion/models/partner_compassion.py

```diff
@@ -522,6 +529,59 @@ def action_view_partner_invoices(self):
         action["context"] = {"search_default_partner_id": self.id}
         return action
 
+    def check_phone_and_mobile(self, vals):
+        all_phone_destination_codes = [
+            21,
+            22,
+            24,
+            26,
+            27,
+            31,
+            32,
+            33,
+            34,
+            41,
+            43,
+            44,
+            51,
+            52,
+            55,
+            56,
+            58,
+            61,
+            62,
+            71,
+        ]
+
+        all_mobile_destination_codes = [74, 75, 76, 77, 78, 79]
+
+        phone = vals.get("phone")
+        mobile = vals.get("mobile")
+
+        if phone:
+            parsed_phone = phonenumbers.parse(phone, "CH")
+            if not phonenumbers.is_valid_number(parsed_phone):
+                raise UserError(_("Phone number is not valid."))
+            phone_national_destination_code = int(str(parsed_phone.national_number)[:2])
+            if phone_national_destination_code in all_mobile_destination_codes:
+                vals["mobile"] = phone
+                if not mobile:
+                    vals["phone"] = False
+
+        if mobile:
+            parsed_mobile = phonenumbers.parse(mobile, "CH")
+            if not phonenumbers.is_valid_number(parsed_mobile):
+                raise UserError(_("Mobile number is not valid."))
+            mobile_national_destination_code = int(
+                str(parsed_mobile.national_number)[:2]
+            )
+            if mobile_national_destination_code in all_phone_destination_codes:
+                vals["phone"] = mobile
+                if not phone:
+                    vals["mobile"] = False
```

Comment:
Maybe we should better check if a `phone` value is already set and not override it, and also check that we didn't move the `phone` field value and avoid deleting it afterward (you should introduce the bool var for that purpose).
```suggestion
                if not vals.get("phone"):
                    vals["phone"] = mobile
                    if not phone_moved_to_mobile:
                        vals["mobile"] = False
```